src/OrbitBase/CanceledOrTest.cpp

```diff
@@ -13,6 +14,34 @@
 
 namespace orbit_base {
 
+TEST(CanceledOr, Construct) {
+  CanceledOr<void> void_or_canceled{};
+  EXPECT_TRUE(void_or_canceled.HasValue());
+  EXPECT_FALSE(void_or_canceled.IsCanceled());
+
+  void_or_canceled = Canceled{};
+  EXPECT_FALSE(void_or_canceled.HasValue());
+  EXPECT_TRUE(void_or_canceled.IsCanceled());
+
+  CanceledOr<int> int_or_canceled{42};
+  EXPECT_TRUE(int_or_canceled.HasValue());
+  EXPECT_FALSE(int_or_canceled.IsCanceled());
+  EXPECT_EQ(int_or_canceled.GetValue(), 42);
+
+  int_or_canceled = Canceled{};
+  EXPECT_FALSE(int_or_canceled.HasValue());
+  EXPECT_TRUE(int_or_canceled.IsCanceled());
+
+  CanceledOr<std::unique_ptr<int>> unique_int_or_canceled{std::make_unique<int>(42)};
+  EXPECT_TRUE(unique_int_or_canceled.HasValue());
+  EXPECT_FALSE(unique_int_or_canceled.IsCanceled());
+  EXPECT_THAT(unique_int_or_canceled.GetValue(), testing::Pointee(42));
```

Comment:
This resolves to `GetValue() const&`. This way the other two overloads remain untested. TBH, I don't care much about `GetValue() const&&`, but I'd like to see `GetValue()&&` tested.

Reply:
Also, just a suggestion. We will have the same tests of these methods for `NotFoundOr`. How about a type parameterised test?

Reply:
This wouldn't be a type parametrized test it would be a template-parametrized test and this is not supported by gtest. Of course you could have `CanceledOr<int>, NotFoundOr<int>, CanceledOr<void>, ...` as types but this just gets a giant mess of `if constexpr`. I think prefer the current way - even with more redundancy.

The GetValue aspect is very valid though. I will fix that. Thank you!

Reply:
Done.